cmd/context.go

```diff
@@ -0,0 +1,225 @@
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/CircleCI-Public/circleci-cli/api"
+	"github.com/CircleCI-Public/circleci-cli/client"
+	"github.com/CircleCI-Public/circleci-cli/git"
+	"github.com/olekukonko/tablewriter"
+	"github.com/pkg/errors"
+
+	"github.com/CircleCI-Public/circleci-cli/settings"
+	"github.com/spf13/cobra"
+)
+
+type Organization struct {
+	name     string
+	provider git.VcsType
+}
+
+// TODO: initialize the client in preRun
+// This will allow us to fill in values passed on the command line like --debug
+
+func newContextCommand(config *settings.Config) *cobra.Command {
+
+	var organization Organization
+	if vcs, org, err := git.InferOrganizationFromGitRemotes(); err == nil {
+		organization.name = org
+		organization.provider = vcs
+	}
+
+	command := &cobra.Command{
+		Use:   "context",
+		Short: "Contexts provide a mechanism for securing and sharing environment variables across projects. The environment variables are defined as name/value pairs and are injected at runtime.",
+	}
+
+	client := client.NewClient(config.Host, config.Endpoint, config.Token, true) //config.Debug)
+
+	listCommand := &cobra.Command{
+		Short: "List contexts",
+		Use:   "list",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return listContexts(client, organization)
+		},
+	}
+
+	showContextCommand := &cobra.Command{
+		Short: "Show a context",
+		Use:   "show <name>",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return showContext(client, organization, args[0])
+		},
+		Args: cobra.ExactArgs(1),
+	}
+
+	storeCommand := &cobra.Command{
+		Short: "Store an new secret in the named context. The value is read from stdin.",
+		Use:   "store <context name> <variable name>",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return storeEnvVar(client, organization, args[0], args[1])
+		},
+		Args: cobra.ExactArgs(2),
+	}
+
+	removeCommand := &cobra.Command{
+		Short: "Remove a secret from the named context",
+		Use:   "remove <context name> <variable name>",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return removeEnvVar(client, organization, args[0], args[1])
+		},
+		Args: cobra.ExactArgs(2),
+	}
+
+	createContextCommand := &cobra.Command{
+		Short: "Create a new context",
+		Use:   "create <context name>",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return createContext(client, organization, args[0])
+		},
+		Args: cobra.ExactArgs(1),
+	}
+
+	deleteContextCommand := &cobra.Command{
+		Short: "Delete the named context",
+		Use:   "delete <context name>",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deleteContext(client, organization, args[0])
+		},
+		Args: cobra.ExactArgs(1),
+	}
+
+	command.PersistentFlags().StringVar(&organization.name, "organization", organization.name, "The organization to operate on.")
```

Comment:
How does `organization.provider` get set? Or does it not matter for the contexts api?